rust/noosphere-p2p/src/dht/types.rs

```diff
@@ -0,0 +1,130 @@
+use crate::dht::channel::{Message, MessageClient, MessageProcessor};
+use crate::dht::errors::DHTError;
+use libp2p::swarm::NetworkInfo;
+use std::{fmt, str};
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum DHTStatus {
+    Active,
+    Terminated,
+    Error(String),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct DHTNetworkInfo {
+    pub num_peers: usize,
+    pub num_connections: u32,
+    pub num_pending: u32,
+    pub num_established: u32,
+}
+
+impl From<NetworkInfo> for DHTNetworkInfo {
+    fn from(info: NetworkInfo) -> Self {
+        let c = info.connection_counters();
+        DHTNetworkInfo {
+            num_peers: info.num_peers(),
+            num_connections: c.num_connections(),
+            num_pending: c.num_pending(),
+            num_established: c.num_established(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum DHTRequest {
+    Bootstrap,
+    //WaitForPeers(usize),
```

Comment:
Is this for the case where you want to ensure a peer quorum of some size?

Reply:
In lieu of "events" (messages from processor to client), this was a work around to get tests just running with some form of confidence. I think "events" would remove the need for this (with maybe the client side still having a helper function for it, maybe just in tests, built off of said Events). Other DHT commands like putting records can have a quorum constraint as well